certbot/tests/log_test.py

```diff
@@ -27,12 +26,33 @@ def _call(cls, *args, **kwargs):
         return pre_arg_parse_setup(*args, **kwargs)
 
     def test_it(self):
-        with mock.patch('certbot.log.except_hook') as mock_except_hook:
-            with mock.patch('certbot.log.sys') as mock_sys:
-                self._call()
-
+        with mock.patch('certbot.log.util.atexit_register') as mock_register:
```

Comment:
Haha. I made this change, but yeah Python doesn't have a problem with it. The scope of variables is only limited to functions, classes, and modules. So for example, all of these things are valid:
```
def foo():
    if condition:
        var = True
    else:
        var = False
    print var 
```
```
with open('foo.txt', 'w') as f:
    f.write('hi')
print f
```
```
nums = [i for i in range(10)]
print i
```
This is strange behavior for people familiar with other languages like C/C++ though so we probably shouldn't do it.